alchemiscale/tests/integration/storage/test_statestore.py

```diff
@@ -1324,6 +1327,52 @@ def test_claim_taskhub_tasks(self, n4js: Neo4jStore, network_tyk2, scope_test):
         claimed6 = n4js.claim_taskhub_tasks(taskhub_sk, csid, count=2)
         assert claimed6 == [None] * 2
 
+    def test_claim_taskhub_tasks_protocol_split(
+        self, n4js: Neo4jStore, network_tyk2, scope_test
+    ):
+        an = network_tyk2
+        network_sk, taskhub_sk, _ = n4js.assemble_network(an, scope_test)
+
+        def reducer(collection, transformation):
+            protocol = transformation.protocol.__class__
+            if len(collection[protocol]) >= 3:
+                return collection
+            sk = n4js.get_scoped_key(transformation, scope_test)
+            collection[transformation.protocol.__class__].append(sk)
+            return collection
+
+        transformations = reduce(
+            reducer,
+            an.edges,
+            {DummyProtocolA: [], DummyProtocolB: [], DummyProtocolC: []},
+        )
```

Comment:
Would this all be about the same as using a `defaultdict(list)` and adding each `Transformation` `ScopedKey` up to a max count for each key? This form is fine, but I probably would have expressed it with a `defaultdict` and loop instead.

Reply:
Happy to keep it! You taught me something new as well; wanted to make sure I was understanding it right. :grin: